source/core/MicroBitFiber.cpp

```diff
@@ -475,7 +523,7 @@ int invoke(void (*entry_fn)(void))
     if (!fiber_scheduler_running())
 		return MICROBIT_NOT_SUPPORTED;
 
-    if (currentFiber->flags & MICROBIT_FIBER_FLAG_FOB)
+    if (currentFiber->flags & (MICROBIT_FIBER_FLAG_FOB | MICROBIT_FIBER_FLAG_PARENT | MICROBIT_FIBER_FLAG_CHILD) || HAS_THREAD_USER_DATA)
```

Comment:
@mmoskal Can you please describe the conditions when HAS_THREAD_USER_DATA is set? Will this affect all MakeCode application fibers / event handlers? Wouldn't this affect RAM efficiency and context switch time somewhat?

I also wonder if a global CONFIG option to disable FOB might be simpler if that's the case? Just thinking of future maintenance here.  Would be good to get your thoughts.